tls/s2n_encrypted_extensions.c

```diff
@@ -118,20 +118,13 @@ int s2n_server_encrypted_extensions_parse(struct s2n_connection *conn, struct s2
         case TLS_EXTENSION_ALPN:
             GUARD(s2n_recv_server_alpn(conn, &extension));
             break;
-        case TLS_EXTENSION_SCT_LIST:
-            GUARD(s2n_recv_server_sct_list(conn, &extension));
-            break;
         case TLS_EXTENSION_MAX_FRAG_LEN:
             GUARD(s2n_recv_server_max_fragment_length(conn, &extension));
             break;
         /* Error on known extensions that are not supposed to appear in EE
          * https://tools.ietf.org/html/rfc8446#page-37
          */
-        case TLS_EXTENSION_RENEGOTIATION_INFO:
-        case TLS_EXTENSION_STATUS_REQUEST:
-        case TLS_EXTENSION_SESSION_TICKET:
-        case TLS_EXTENSION_SUPPORTED_VERSIONS:
-        case TLS_EXTENSION_KEY_SHARE:
+        default:
```

Comment:
previously, we accepted 4 extensions, blacklisted 5 extensions, ignored unknown extensions. with this change, we are accepting only 3 extensions and erroring on the rest.

i think we should keep to the previous behavior of listing blacklisted extensions, unknown values such as new or greased extensions can be safely ignored without effects.

Reply:
sure we can do that